src/transformer/repos/TestRepository.php

```diff
@@ -48,4 +48,15 @@ public function read_records($type, array $query) {
 
         return $matchingrecords;
     }
+
+    /**
+     * Reads records in a pure sql format. Used for quiz attempt lookup.
+     * @param String $sql
+     * @return mixed|PhpArr
+     */
+    public function read_records_sql($sql) {
```

Comment:
I think this is likely to cause problems in future developments. Take a loot at one of my earlier comments where I suggest using `read_record` instead. I know it's extra DB calls, but it will allow those other records to be used in future. The extra DB calls should be pretty quick and ran in via the cron anyway so the additional time is quite negligible.

Reply:
removed as requested in favor of multiple lookups.